src/Test.FunctionalTests.Sarif/DefaultObjectPopulatingVisitor.cs

```diff
@@ -195,7 +195,8 @@ private void PopulatePropertyWithGeneratedDefaultValue(ISarifNode node, Property
             object propertyValue = null;
             Type propertyType = property.PropertyType;
 
-            bool isRequired = PropertyIsRequiredBySchema(node.GetType().Name, property.Name);
+            bool isRequired = PropertyIsRequiredBySchema(node.GetType().Name, property.Name) || 
+                PropertyIsAnyOfRequiredBySchema(node.GetType().Name, property.Name);
```

Comment:
>PropertyIsAnyOfRequiredBySchema [](start = 16, length = 31)

This logic doesn't look right. If a property appears in an `anyOf`, as `b` does here...
```
"someObject": {
  "required": [ "a" ],
  "anyOf": [
    {
     "required": [ "b" ]
    },
    {
      "required": [ "c" ]
    }
  ]
}
```
... you wouldn't conclude that `b` is required.

I see your comment below that this is a "quick fix" for something -- that at least for now, you must treat something as required if it appears in any `anyOf` element -- but why? Maybe I'll find out later...

Reply:
Anyway, I'm not really supposed to be reviewing SDK code, I should only be looking at the schema, so, over to that...

---
In reply to: [275125189](https://github.com/Microsoft/sarif-sdk/pull/1426#discussion_r275125189) [](ancestors = 275125189)

Reply:
Closing loop: Harleen explained this to me. The variable name `isRequired` might be better, but I understand that the idea is to ensure that you don't end up generating a SARIF file that's missing a required property (or that has _none_ of a set of alternative required properties).

---
In reply to: [275125217](https://github.com/Microsoft/sarif-sdk/pull/1426#discussion_r275125217) [](ancestors = 275125217,275125189)

Reply:
Thanks Larry, as discussed, i am going to add a comment to clearly explain the intention. I like the way you described it so i might plagiarize this.

---
In reply to: [275450923](https://github.com/Microsoft/sarif-sdk/pull/1426#discussion_r275450923) [](ancestors = 275450923,275125217,275125189)